<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" rowspan="2">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Simulation</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="RenderArea" name="render_area" native="true">
         <property name="minimumSize">
          <size>
           <width>525</width>
           <height>475</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">background: rgb(0, 0, 0);</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="toggle_simulation">
           <property name="text">
            <string>Start</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="reset_simulation">
           <property name="text">
            <string>Reset</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="sim_time_slider">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>10</number>
           </property>
           <property name="pageStep">
            <number>10</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="tickPosition">
            <enum>QSlider::TicksAbove</enum>
           </property>
           <property name="tickInterval">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="minimumSize">
       <size>
        <width>400</width>
        <height>284</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>400</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Forward Kinematics</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Left Wheel Speed (rev/s)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="left_wheel_spinbox">
         <property name="minimum">
          <number>-10</number>
         </property>
         <property name="maximum">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="left_wheel_slider">
         <property name="minimum">
          <number>-10</number>
         </property>
         <property name="maximum">
          <number>10</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksAbove</enum>
         </property>
         <property name="tickInterval">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Right Wheel Speed (rev/s)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="right_wheel_spinbox">
         <property name="minimum">
          <number>-10</number>
         </property>
         <property name="maximum">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="right_wheel_slider">
         <property name="minimum">
          <number>-10</number>
         </property>
         <property name="maximum">
          <number>10</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksAbove</enum>
         </property>
         <property name="tickInterval">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>400</width>
        <height>284</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>400</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Inverse Kinematics</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Linear Velocity (m/s)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="linear_velocity_spinbox">
         <property name="minimum">
          <number>-10</number>
         </property>
         <property name="maximum">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="linear_velocity_slider">
         <property name="minimum">
          <number>-10</number>
         </property>
         <property name="maximum">
          <number>10</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksAbove</enum>
         </property>
         <property name="tickInterval">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Angular Velocity (m/s)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="angular_velocity_spinbox">
         <property name="minimum">
          <number>-10</number>
         </property>
         <property name="maximum">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="angular_velocity_slider">
         <property name="minimum">
          <number>-10</number>
         </property>
         <property name="maximum">
          <number>10</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksAbove</enum>
         </property>
         <property name="tickInterval">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RenderArea</class>
   <extends>QWidget</extends>
   <header>renderarea</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>left_wheel_slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>left_wheel_spinbox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>787</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>787</x>
     <y>69</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>left_wheel_spinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>left_wheel_slider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>787</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>787</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>right_wheel_slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>right_wheel_spinbox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>787</x>
     <y>204</y>
    </hint>
    <hint type="destinationlabel">
     <x>787</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>right_wheel_spinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>right_wheel_slider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>787</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>787</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
